src/lib/utils/npm-user-agent-parser.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Parses a user agent string for NPM metadata.
+ * @param {string} userAgent 
+ * @return {{npmVersion: string, nodeVersion: string, platform: string, architecture: string}}
+ */
+const npmUserAgentParser = (userAgent) => {
```

Comment:
From a cursory google I couldn't find an existing library to parse NPM user agent strings, standard user agent parsers don't appear to work with NPM's user agent format.

Would a utility library under verdaccio scope be created in this git repository or a separate one?